### ISSUE TITLE
`torch.float` causing a seg fault

### ISSUE THREAD

**Issue by miladm** [pytorch/xla#3380]
## 🐛 Bug

Running into this Seg Fault issue when using `torch.float` tpye in the following code snippet:

```
> export XLA_EXPERIMENTAL="nonzero:masked_select"
> python3
```

```
>>> import torch, torch_xla
>>> import torch_xla.core.xla_model as xm
>>> dev = xm.xla_device()

>>> a1 = torch.tensor([[1,0,0,5,0,6]], device=dev)
>>> a2 = torch.nonzero(a1)

>>> a3 = torch.cat((a2,a2),dim=1).type(torch.float)
>>> a3.shape
torch.Size([6, 4])

>>> torch_xla._XLAC._get_xla_tensor_dimension_size(a3,0)
https://symbolize.stripped_domain/r/?trace=7f721bc0b04c,7f722f4bc3bf&map=
*** SIGSEGV (@0x8), see gl__________41#s15 received by PID 840736 (TID 840736) on cpu 74; stack trace: ***
PC: @     0x7f721bc0b04c  (unknown)  c10::impl::ExcludeDispatchKeyGuard::~ExcludeDispatchKeyGuard()
    @     0x7f71673bb7fa        992  (unknown)
    @     0x7f722f4bc3c0  (unknown)  (unknown)
https://symbolize.stripped_domain/r/?trace=7f721bc0b04c,7f71673bb7f9,7f722f4bc3bf&map=7a511a57244151c993b16b37978e7ed7:7f71589e3000-7f7167725d50
E0216 16:59:35.045972  840736 coredump_hook.cc:365] RAW: Remote crash data gathering hook invoked.
E0216 16:59:35.045993  840736 coredump_hook.cc:411] RAW: Skipping coredump since rlimit was 0 at process start.
E0216 16:59:35.046004  840736 client.cc:221] RAW: Coroner client retries enabled (b/136286901), will retry for up to 30 sec.
E0216 16:59:35.046012  840736 coredump_hook.cc:473] RAW: Sending fingerprint to remote end.
E0216 16:59:35.046025  840736 coredump_socket.cc:124] RAW: Stat failed errno=2 on socket /var/google/services/logmanagerd/remote_coredump.socket
E0216 16:59:35.046037  840736 coredump_hook.cc:477] RAW: Cannot send fingerprint to Coroner: [NOT_FOUND] Missing crash reporting socket. Is the listener running?
E0216 16:59:35.046047  840736 coredump_hook.cc:550] RAW: Discarding core.
E0216 16:59:35.262469  840736 process_state.cc:770] RAW: Raising signal 11 with default behavior
Segmentation fault (core dumped)
```

With `a3 = torch.cat((a2,a2),dim=1)`, the code functions correctly.

**Comment by miladm** [comment]
This issue happens on `a2`, most likely because this tensor is dynamic and dynamic shape is not fully supported yet.
